### PR TITLE
support aliasing root fields and column fields in console; introduce ReasonML in console codebase

### PR DESCRIPTION
**Pull request by wawhal**
Fixes internal issue-270.
Implements the UI for setting GraphqL aliases to your root fields and columns.
Also introduces ReasonML in the codebase.

### Affected components 

- Console

### CODE REVIEW

**Comment by marionschleifer** [comment]
@wawhal just asking to confirm that no additional documentation is needed for this?

File: server/src-exec/Migrate.hs [hunk 1]
```diff
@@ -432,4 +432,4 @@ migrateCatalog migrationTime = migrateFrom =<< getCatalogVersion
                 UPDATE "hdb_catalog"."hdb_version"
                    SET "version" = $1,
                        "upgraded_on" = $2
-                    |] (curCatalogVer, migrationTime) False
```
Comment (lexi-lambda) [inline server/src-exec/Migrate.hs]: I’m not sure why a bunch of these files in the server code had their trailing newlines stripped, but could you revert these changes on your branch? (That way I won’t be a code owner on this PR, and you won’t have to wait for me to review it.)

Comment (wawhal) [reply server/src-exec/Migrate.hs]: Done.

**Comment by wawhal** [comment]
This PR is re-done [here](#3175).
